### PR TITLE
Add grub2_bootloader_argument_absent template and nosmep/nosmap rules

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Based on `grub2_bootloader_argument`, introduce `grub2_bootloader_argument_absent`
  - The template ensure that an argument is not specified in the kernel command line
- Adds two rules that check if security features are not disabled in grub2 command line
  - `grub2_nosmep_argument_absent` - Ensures that SMEP is not disabled during boot time
  - `grub2_nosmap_argument_absent` - Ensures that SMAP is not disabled during boot time

#### Rationale:

- These 2 rules checks whether default security features are not disabled at boot time.

### CODE REVIEW

**Comment by yuumasato** [comment]
Should we force existence of `/etc/default/grub` in our test scenarios so that the tests work in containers?
```
/ssgts/shared bash -x arg_there_etcdefaultgrub.fail.sh' on root@localhost is 2: Warning: Permanently added '[localhost]:41547' (ECDSA) to the list of known hosts.
+ grep -q '^GRUB_CMDLINE_LINUX=.*nosmep.*"' /etc/default/grub
grep: /etc/default/grub: No such file or directory
+ sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)"$/\1 nosmep"/' /etc/default/grub
sed: can't read /etc/default/grub:
```

**Comment by jan-cerny** [comment]
Ideally yes, I would love to run as many tests in GitHub CI as possible instead of having to run them locally, try to add `# packages = grub2-tools` to scenarios and we'll see what will happen.

File: linux_os/guide/system/bootloader-grub2/grub2_nosmap_argument_absent/rule.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+documentation_complete: true
+
+title: 'Ensure SMAP is not disabled during boot'
+
+description: |-
+    The SMAP is used to prevent the supervisor mode from unintentionally reading/writing into
+    memory pages in the user space, it is enabled by default since Linux kernel 3.7.
+    But it could be disabled through kernel boot parameters.
+
+    Ensure that Supervisor Mode Access Prevention (SMAP) is not disable by
```
Comment (jan-cerny) [inline linux_os/guide/system/bootloader-grub2/grub2_nosmap_argument_absent/rule.yml:10]:
```suggestion
    Ensure that Supervisor Mode Access Prevention (SMAP) is not disabled by
```

**Review by jan-cerny — COMMENTED** [review]
otherwise looks great

File: linux_os/guide/system/bootloader-grub2/grub2_nosmap_argument_absent/rule.yml (hunk 1 above)
Comment (yuumasato) [reply linux_os/guide/system/bootloader-grub2/grub2_nosmap_argument_absent/rule.yml]: Fixed by rebase in 4cc98d2bd6098f214cd4c5af012657067ae7378c

File: linux_os/guide/system/bootloader-grub2/grub2_nosmep_argument_absent/rule.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+documentation_complete: true
+
+title: 'Ensure SMEP is not disabled during boot'
+
+description: |-
+    The SMEP is used to prevent the supervisor mode from executing user space code,
+    it is enabled by default since Linux kernel 3.0. But it could be disabled through
+    kernel boot parameters.
+
+    Ensure that Supervisor Mode Execution Prevention (SMEP) is not disable by
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply linux_os/guide/system/bootloader-grub2/grub2_nosmep_argument_absent/rule.yml]: Fixed by rebase in 9004413d780f1a72bd6907c57b645d29b75cd05e

**Comment by yuumasato** [comment]
> Ideally yes, I would love to run as many tests in GitHub CI as possible instead of having to run them locally, try to add `# packages = grub2-tools` to scenarios and we'll see what will happen.

I have also added `grubby` package, so that the remediation tool is available.

**Comment by jan-cerny** [comment]
The problem seems to be that even if the package is installed, the files aren't actually created. Nevertheless, the test scenarios pass locally:

```
[jcerny@thinkpad scap-security-guide{pr/8259}]$ python3 tests/test_suite.py rule --libvirt qemu:///system ssgts_rhel8 grub2_nosmap_argument_absent
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-02-28-1413/test_suite.log
WARNING - Script mising_arg_invalid_rescue.pass.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_grub2_nosmap_argument_absent
INFO - Script arg_there_grubenv.fail.sh using profile (all) OK
INFO - Script arg_value_there_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script missing_arg.pass.sh using profile (all) OK
INFO - Script superstring_arg.pass.sh using profile (all) OK
INFO - Script arg_there_etcdefaultgrub.fail.sh using profile (all) OK
```

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**